client/components/SingleArtwork.js

```diff
@@ -1,49 +1,69 @@
 import React from  'react';
 import { connect } from 'react-redux';
 import { getArtwork } from '../redux/artwork';
-import {Link} from 'react-router-dom'
-
+import { Link } from 'react-router-dom'
+import { addCartItem } from '../redux/cart'
 
 export class SingleArtwork extends React.Component {
+    constructor() {
+        super()
+        this.addQty = this.addQty.bind(this)
+    }
+
+    async addQty(){
+        const quantity = +document.getElementById('qty').value
+        const artworkId = this.props.artwork.id
+        const cartItem = { artworkId, quantity }
+        //add qty to DB if logged in
+        let isLoggedIn = false
+        if(this.props.user.id) isLoggedIn = true
+        this.props.addCartItem(cartItem, isLoggedIn)
+    }
 
     async componentDidMount(){
         await this.props.getArtwork(this.props.match.params.id)
     }
 
     render(){
-        console.log("props in render", this.props.artwork)
+        const quantitySelection = []
+        for (let i = 1; i <= this.props.artwork.quantity; i++) {
+            quantitySelection.push(<option key={i} value={i}>{i}</option>)
+        }
```

Comment:
i understand the logic here, one thing i'm considering is that for artwork that has A LOT in stock (for ex, one piece in the seed file has a quantity over 700) i'm not sure if it makes sense for this to be a dropdown. this is more so business logic, but if hypothetically people aren't mass buying art then we can probably just limit it to 10 or quantity, whichever is less.

Reply:
Yeah I thought about that, too. I think the way you describe it (drop down to 10, manually enter >10) is the way Amazon does it. Maybe for the purposes of this class, we should limit our inventory to 10...